src/allmydata/test/test_istorageserver.py

```diff
@@ -1046,18 +1091,87 @@ def disconnect(self):
         assert self.storage_client is not current
 
 
+class _HTTPMixin(_SharedMixin):
+    """Run tests on the HTTP version of ``IStorageServer``."""
+
+    def setUp(self):
+        if PY2:
+            self.skipTest("Not going to bother supporting Python 2")
+        return _SharedMixin.setUp(self)
+
+    @inlineCallbacks
+    def _get_istorage_server(self):
+        swissnum = b"1234"
+        http_storage_server = HTTPServer(self.server, swissnum)
+
+        # Listen on randomly assigned port:
+        tcp_address, endpoint_string = self._port_assigner.assign(reactor)
+        _, host, port = tcp_address.split(":")
+        port = int(port)
+        endpoint = serverFromString(reactor, endpoint_string)
+        listening_port = yield endpoint.listen(Site(http_storage_server.get_resource()))
+        self.addCleanup(listening_port.stopListening)
+
+        # Create HTTP client with non-persistent connections, so we don't leak
+        # state across tests:
+        treq_client = HTTPClient(
+            Agent(reactor, HTTPConnectionPool(reactor, persistent=False))
+        )
```

Comment:
I have a vague doubt about whether `persistent=False` is the only thing required to make sure `Agent` / `HTTPConnectionPool` doesn't leak selectables past the end of a test method but I can't easily substantiate it ... I guess the easiest thing to do is just wait and see and deal with it if it happens.